src/Microsoft.AspNetCore.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -162,6 +160,14 @@ public OAuthHandler(HttpClient backchannel)
             var ticket = new AuthenticationTicket(new ClaimsPrincipal(identity), properties, Options.AuthenticationScheme);
             var context = new OAuthCreatingTicketContext(ticket, Context, Options, Backchannel, tokens);
             await Options.Events.CreatingTicket(context);
+
+            if (ticket.Principal.FindFirst(ClaimTypes.NameIdentifier) == null)
+            {
+                // If there aren't any claims at all, at least add a unique id for this login session. This will help associate the
+                // tokens from the token store.
+                identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, Guid.NewGuid().ToString(), ClaimValueTypes.String, Options.ClaimsIssuer));
```

Comment:
We're clearly opening Pandora's box here. Generating a unique per-login identifier would be misleading for application code and may result in terrible things like being able to associate a provider with multiple Identity accounts.